### PR TITLE
Use Github CI in addition to Travis

### LINKED ISSUE #385: Set up Github CI as an alternative to Travis CI
We can still keep Travis around and enable allowed failures for all the platforms we're testing there, but we should consider adding Github CI through its Actions platform.

Benefits:
- Inline log view in the Actions panel.
- Builds are started very quickly. From push to testing it takes only a couple of seconds.
- It should be possible to set up an `auto-merge` action for approved PRs, which improves workflow.
- Faster network I/O and hardware.

Drawbacks:
- Support for D is still experimental (but this is not a blocker at all)
- Actions are still a Beta feature. The final roll-out is expected in November.
- Faster hardware could mean timing-related bugs become latent and hidden. But this is an orthogonal issue, and simulating slower hardware should be possible.

Example runs:

mac OS w/ DMD: https://github.com/AndrejMitrovic/agora/commit/3168622a8a82581176a3b7a0aa11fa047e7aa650/checks?check_suite_id=266857679

mac OS w/ LDC:
https://github.com/AndrejMitrovic/agora/commit/a33ffd2c6c11f80028c974c45b0aff6092882231/checks?check_suite_id=266867618

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
Fixes https://github.com/bpfkorea/agora/issues/385

### CODE REVIEW

**Comment by AndrejMitrovic** [comment]
Btw, I couldn't figure out how to have OS-specific commands. The only way it seems doable is with `runs-on`. But this applies to jobs, and jobs are ran in parallel. So I had to split these into two job definitions.

File: .github/workflows/github.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      with:
+        submodules: true
+    - name: Prepare compiler
+      uses: mihails-strasuns/setup-dlang@v0.3.0
+      with:
+          compiler: ${{ matrix.dc }}
+    - name: Install Dub & Sqlite3
+      run: |
+        ## BUG: Must install dub via brew, the one installed via setup-dlang
+        ## doesn't link properly: libsqlite3 linker errors
```
Comment (Geod24) [inline .github/workflows/github.yml]: Have you tried installing only `pkg-config` ?

File: .github/workflows/github.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+name: Github
+on: [push, pull_request]
+
+jobs:
+  test_mac:
+    name: MacOS
+    strategy:
+      matrix:
+        os: [macOS-10.14]
+        dc: [dmd-2.088.1, dmd-2.087.1, ldc-1.17.0, ldc-1.16.0, dmd-master, ldc-master]
```
Comment (Geod24) [inline .github/workflows/github.yml]:
```suggestion
        dc: [dmd-2.088.1, dmd-2.087.1, ldc-1.18.0, ldc-1.17.0, dmd-master, ldc-master]
```

**Comment by Geod24** [comment]
Sweet!

**Comment by AndrejMitrovic** [comment]
Trying out `pkg-config` & newer ldc.

**Comment by Geod24** [comment]
Looks like it worked

**Review by Geod24 — APPROVED** [review]

**Comment by Geod24** [comment]
Very noice

**Pull request merged**